### PR TITLE
improve tempo

### PR DESCRIPTION
**Pull request by rettinghaus**
This PR fixes tempo alignement. Before it always was attached to first glyph of the measure.

### CODE REVIEW

**Comment by rettinghaus** [comment]
Before:
tempo-attachment-old
After:
tempo-attachment

TESTFILE:
```
<?xml version="1.0" encoding="UTF-8"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="4.0.0">
    <meiHead>
        <fileDesc>
            <titleStmt>
                <title>Test tempo attachment</title>
            </titleStmt>
            <pubStmt/>
        </fileDesc>
    </meiHead>
    <music>
        <body>
            <mdiv>
                <score>
                    <scoreDef>
                        <staffGrp>
                            <staffDef clef.shape="G" clef.line="2" meter.count="4" meter.unit="4" n="1" lines="5"/>
                        </staffGrp>
                    </scoreDef>
                    <section>
                        <pb/>
                        <measure n="1">
                            <staff n="1">
                                <layer n="1">
                                    <rest dur="4"/>
                                    <rest dur="4"/>
                                    <rest dur="4"/>
                                    <rest dur="4"/>
                                </layer>
                            </staff>
                            <tempo tstamp="1">Tempo I</tempo>
                        </measure>
                        <measure n="2">
                            <staff n="1">
                                <layer n="1">
                                    <rest dur="4"/>
                                    <rest dur="4" xml:id="rest2"/>
                                    <rest dur="4"/>
                                    <rest dur="4"/>
                                </layer>
                            </staff>
                            <tempo startid="#rest2">Tempo II</tempo>
                        </measure>
                        <measure n="3">
                            <staff n="1">
                                <layer n="1">
                                    <rest dur="4"/>
                                    <rest dur="4"/>
                                    <rest dur="4"/>
                                    <rest dur="4"/>
                                </layer>
                            </staff>
                            <tempo tstamp="3">Tempo III</tempo>
                        </measure>
                        <measure n="4">
                            <staff n="1">
                                <layer n="1">
                                    <rest dur="4"/>
                                    <rest dur="4"/>
                                    <rest dur="4"/>
                                    <rest dur="4"/>
                                </layer>
                            </staff>
                        </measure>
                    </section>
                </score>
            </mdiv>
        </body>
    </music>
</mei>
```

File: src/view_control.cpp [hunk 1]
```diff
@@ -1792,20 +1792,14 @@ void View::DrawTempo(DeviceContext *dc, Tempo *tempo, Measure *measure, System *
...
-        alignmentComparison.SetType(ALIGNMENT_DEFAULT);
-        pos = dynamic_cast<Alignment *>(measure->m_measureAligner.FindDescendantByComparison(&alignmentComparison, 1));
-    }
-    // if we found one, use it
-    if (pos) {
-        params.m_x += pos->GetXRel();
+    params.m_x = tempo->GetStart()->GetDrawingX();
+    if (tempo->HasTstamp()) {
+        params.m_x -= 2 * tempo->GetStart()->GetDrawingRadius(m_doc);
+        if ((tempo->GetTstamp() <= 1) && pos) params.m_x = measure->GetDrawingX() + pos->GetXRel();
```
Comment (lpugin) [inline src/view_control.cpp]: You should flip the two lines and make them if / else

Comment (rettinghaus) [reply src/view_control.cpp]: done

File: src/view_control.cpp [hunk 2]
```diff
@@ -1792,20 +1792,16 @@ void View::DrawTempo(DeviceContext *dc, Tempo *tempo, Measure *measure, System *
...
-    }
-    // if we found one, use it
-    if (pos) {
-        params.m_x += pos->GetXRel();
+    params.m_x = tempo->GetStart()->GetDrawingX();
+    if (!tempo->HasStartid()) {
+        if ((tempo->GetTstamp() <= 1) && pos)
+            params.m_x = measure->GetDrawingX() + pos->GetXRel();
+        else
+            params.m_x -= 2 * tempo->GetStart()->GetDrawingRadius(m_doc);
```
Comment (lpugin) [inline src/view_control.cpp]: Thanks. FYI, I am now avoid if / else statements without {} See https://github.com/rism-ch/verovio/blob/master/doc/guidelines.md#exception-for-one-line-control-clauses
The only exception remains with only if statements (without else) that can be written on a single line

**Comment by wergo** [comment]
Thanks for this! I wanted to raise this as issue earlier...

File: src/view_control.cpp (hunk 2 above)
Comment (rettinghaus) [reply src/view_control.cpp]: noted

**Pull request merged**